docs/DeveloperGuide.md

```diff
@@ -1,34 +1,678 @@
+  - Pros: User has minimal potential to see unreasonable list in the Shopping List. For 
+  example, having a item that has only price and quantity but without description and also
+  a list that one item is recorded multiple times in the list.
+  
+  - Cons: Will significantly increase the code base as there is a need to check for the 
+  presence of the variable in user input to avoid duplication, not human friendly in certain 
+  scenario (eg: user wants to have duplicates because the item is for different occasion and 
+  the user wants to record down twice without any elaboration).
+
+
+- Alternative 2: Require user to provide all three values to successfully add the item into 
+                 the list. Duplicates are allowed
+
+  - Pros: User will have a neat and unity Shopping list, less deal with parameter (because users are
+  forced to give all three variables). And the duplicates item are useful in certain specific condition.
+  
+  - Cons: User flexibility will decrease, because user must input all parameters even if he/she does not want to 
+  provide certain variables such as price and quantity, which will result unsuccessful adding items into the list. 
+  Also, duplicate items may confused the user, even though in some specific scenario, duplicate item may be useful to 
+  the user.  
+ 
+
+&nbsp;
+<b><a href="#developer-guide">&#129053; back to top</a></b>
+
+&nbsp;
+
+<!-- @@author trishaangelica --> 
+### 3.2 Edit feature
+#### 3.2.1 Current implementation
+
+The edit feature is implemented using an <code>EditCommand</code> class. This class extends from the main
+<code>Command</code> class. The <code>item</code> object to be edited is identified by the index number provided 
+in the user input. In addition to the index no. , the user input **must also contain at least one** of these parameters: 
+*description*, *price*, *quantity*. 
+
+The process of object creation is as follows:
+
+1. <code>Duke</code> class receives user input from the <code>Ui</code> class. 
+2. A <code>Parser</code> object is created to call its <code>parseCommand</code> method.
+* The <code>Parser</code> object instantiates an <code>EditCommand</code> object based on the user input.
+3. The <code>Duke</code> class calls the <code>execute</code> method of the <code>EditCommand</code> object.
+4. In the <code>execute</code> function, the <code>item</code> to be edited (based on the specified index of the 
+user input) is called from the <code>ShoppingList</code> object.The original description/price/quantity of the item is overwritten 
+with the new values from the user input.
+5. The <code>item</code> object with its' new values is stored back to the <code>ShoppingList</code> object.
+
+The following sequence diagram below shows how the edit feature works. The details of updating the items' values
+have been omitted from the diagram. Those details are shown in a separate sequence diagram below:
+
+![alt text](images/EditFeature.png)
```

Comment:
![1](https://user-images.githubusercontent.com/47073395/77987195-c4388000-734b-11ea-87da-84309bd44443.PNG)
I was wondering if the activation bar of duke can just be replaced by a life line, since the activation bar ends before those of the called functions, which may be confusing.